src/index.js

```diff
@@ -5,19 +5,23 @@ import getDevServerUrl from './utils/getDevServerBundleUrl';
 export const getDevServerBundleUrl = getDevServerUrl;
 
 const main = ({ serverConfig, clientConfig }) => {
+  const handles = {};
+
   if (clientConfig) {
     // Start webpack dev server separately on a different port to avoid issues with httpServer restarts
-    watchClientChanges(clientConfig);
+    handles.client = watchClientChanges(clientConfig);
   }
 
   if (serverConfig) {
     // Watch changes on the server side, re-compile and restart.
-    watchServerChanges(serverConfig);
+    handles.server = watchServerChanges(serverConfig);
```

Comment:
I think we can await `watchServerChanges` here to get server.httpServer. Doing so will make both the server and client return types consistent to just Server.

```js
// main
const main = async ({ serverConfig, clientConfig }) => {
  const handles = {};

  if (clientConfig) {
    // Start webpack dev server separately on a different port to avoid issues with httpServer restarts
    handles.client = watchClientChanges(clientConfig);
  }

  if (serverConfig) {
    // Watch changes on the server side, re-compile and restart.
    handles.server = await watchServerChanges(serverConfig);
  }

  return handles;
};

// then the return type of both server and client is just Server
interface UniversalHotReloadHandles {
  server?: Server;
  client?: Server;
}

// and the main function type now returns a Promise
export default function universalHotReload({
  serverConfig,
  clientConfig,
}: UniversalHotReloadConfig): Promise<UniversalHotReloadHandles>;
```

Please update the readme as well to document this usage.